python/lsst/ts/observatory/control/script_queue.py

```diff
@@ -0,0 +1,392 @@
+        script : `str`
+            Script path.
+        config : `dict`
+            Script configuration.
+        description : `str`, optional
+            Short description of why you are running the script
+            (default: empty).
+        log_level : `int`, optional
+            Script log level (default: DEBUG).
+        pause_checkpoint : `str`, optional
+            Pause the script at the specified checkpoint (default: empty).
+        """
+        await self.add(
+            is_standard=True,
+            script=script,
+            config=config,
+            description=description,
+            log_level=log_level,
+            pause_checkpoint=pause_checkpoint,
+        )
+
+    async def add_external(
+        self,
+        script: str,
+        config: dict,
+        description: str = "",
+        log_level: int = logging.DEBUG,
+        pause_checkpoint: str = "",
+    ) -> None:
+        """Add external script to the script queue.
+
+        Parameters
+        ----------
+        script : `str`
+            Script path.
+        config : `dict`
+            Script configuration.
+        description : `str`, optional
+            Short description of why you are running the script
+            (default: empty).
+        log_level : `int`, optional
+            Script log level (default: DEBUG).
+        pause_checkpoint : `str`, optional
+            Pause the script at the specified checkpoint (default: empty).
+        """
+        await self.add(
+            is_standard=False,
+            script=script,
+            config=config,
+            description=description,
+            log_level=log_level,
+            pause_checkpoint=pause_checkpoint,
+        )
+
+    async def add(
+        self,
+        is_standard: bool,
+        script: str,
+        config: dict,
+        description: str = "",
+        log_level: int = logging.DEBUG,
+        pause_checkpoint: str = "",
+    ) -> None:
+        """Add script to the script queue.
+
+        Parameters
+        ----------
+        is_standard : `bool`
+            Is script standard?
+        script : `str`
+            Script path.
+        config : `dict`
+            Script configuration.
+        description : `str`, optional
+            Short description of why you are running the script
+            (default: empty).
+        log_level : `int`, optional
+            Script log level (default: DEBUG).
+        pause_checkpoint : `str`, optional
+            Pause the script at the specified checkpoint (default: empty).
+        """
+        await self.queue_remote.cmd_add.set_start(
+            isStandard=is_standard,
+            path=script,
+            config=yaml.safe_dump(config),
+            descr=description,
+            location=idl.enums.ScriptQueue.Location.LAST,
+            logLevel=log_level,
+            pauseCheckpoint=pause_checkpoint,
+            timeout=self.long_timeout,
+        )
+
+    async def pause(self) -> None:
```

Comment:
Oh, I think there is a confusion about "pausing the ScriptQueue" and "pausing Scripts"... This command here pauses the ScriptQueue only, but if there is a Script running it will continue to run until the end. 

The `pauseCheckpoint` feature is a way to tell Scripts to pause at the next checkpoint. This happens while the ScriptQueue is still running.